src/mechanics/mesh/MeshFem.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#include <boost/ptr_container/ptr_vector.hpp>
+#include "base/Group.h"
+#include "base/ValueVector.h"
+#include "mechanics/DirectionEnum.h"
+#include "mechanics/nodes/NodeSimple.h"
+#include "mechanics/elements/ElementCollection.h"
+
+#include <sstream>
+
+namespace NuTo
+{
+class MeshFem
+{
+public:
+    InterpolationSimple& CreateInterpolation(const InterpolationSimple& interpolation)
+    {
+        mInterpolations.push_back(interpolation.Clone().release());
+        return *mInterpolations.rbegin();
+    }
+
+    //! @brief selects a node of type \p dofType at given \p coords
+    //! @param coords global coordinates
+    //! @param dofType dof type
+    //! @param tol selection tolerance
+    //! @return reference to the selected node, throws, if no node is found
+    NodeSimple& NodeAtCoordinate(Eigen::VectorXd coords, DofType dofType, double tol = 1.e-10)
+    {
+        for (auto& element : this->Elements)
+        {
+            auto& dofElement = element.DofElement(dofType);
+            const auto& dofInterpolation = dofElement.Interpolation();
+            for (int iNode = 0; iNode < dofInterpolation.GetNumNodes(); ++iNode)
+            {
+                NaturalCoords dofNodeCoords = dofInterpolation.GetLocalCoords(iNode);
+                Eigen::VectorXd globalNodeCoords = Interpolate(element.CoordinateElement(), dofNodeCoords);
+                if ((globalNodeCoords - coords).isMuchSmallerThan(tol))
+                    return dofElement.GetNode(iNode);
+            }
+        }
+        std::stringstream coordsString;
+        coordsString << coords.transpose();
+        throw NuTo::Exception(__PRETTY_FUNCTION__,
+                              "There is no node for dof type " + dofType.GetName() + " at " + coordsString.str());
+    }
+
+    //! @brief selects all nodes of type \p dofType where the \p coord in \p direction is within \tol
+    //! @param direction ::X, ::Y, or ::Z
+    //! @param coord coordinate value on the axis
+    //! @param dofType dof type
+    //! @param tol selection tolerance
+    //! @return group with selected nodes, the group may be empty if no nodes were found
+    Groups::Group<NodeSimple> NodeAtAxis(eDirection direction, double coord, DofType dofType, double tol = 1.e-10)
```

Comment:
Small naming thing: I think it should be *Nodes*, because it may return several. Also *Axis* suggests `coord == 0.0` to me, but then again I don't have a good alternative.

Reply:
Jip, i'll fix the naming. What about changing `coord` to `axisOffset = 0`. So it actually has the meaning of an axis, but you can provide an offset.

Reply:
👍